tests/components/home_connect/test_config_flow.py

```diff
@@ -403,3 +403,56 @@ async def test_dhcp_flow_already_setup(
     )
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "already_configured"
+
+
+@pytest.mark.usefixtures("current_request_with_host")
+@pytest.mark.parametrize(
+    ("dhcp_discovery", "appliance"),
+    [
+        (
+            DhcpServiceInfo(
+                ip="1.1.1.1",
+                hostname="bosch-cookprocessor-123456789012345678",
+                macaddress="c8:d7:78:00:00:00",
+            ),
+            "CookProcessor",
+        ),
+        (
+            DhcpServiceInfo(
+                ip="1.1.1.1",
+                hostname="BOSCH-HCS000000-68A40E000000",
+                macaddress="68:a4:0e:00:00:00",
+            ),
+            "Hob",
+        ),
+    ],
+    indirect=["appliance"],
+)
+async def test_dhcp_flow_complete_device_information(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    client: MagicMock,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[MagicMock], Awaitable[bool]],
+    dhcp_discovery: DhcpServiceInfo,
+    appliance: HomeAppliance,
+) -> None:
+    """Test DHCP discovery with complete device information."""
+    assert await integration_setup(client)
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    device = device_registry.async_get_device(identifiers={(DOMAIN, appliance.ha_id)})
+    assert device
+    assert device.connections == set()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_DHCP}, data=dhcp_discovery
+    )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
```

Comment:
Can we test completing a flow to a create entry result too?

Reply:
Sure!

Reply:
https://github.com/home-assistant/core/blob/1d675ab9e6698d39e7511578ddb996c0746316c2/tests/components/home_connect/test_config_flow.py#L341-L390

This is a dhcp flow test that creates an entry, which is already merged

Reply:
Oh, right. 👍